apps/api/src/app/integrations/dtos/credentials.dto.ts

```diff
@@ -50,6 +50,12 @@ export class CredentialsDto implements ICredentials {
   @IsOptional()
   region?: string;
 
+  @ApiPropertyOptional()
+  @TransformToBoolean()
+  @IsBoolean()
+  @IsOptional()
+  eu?: boolean;
```

Comment:
Indeed adding the boolean of this region is a short term fix that will easily become a problem if Sparkpost for example would enable more regions.
I know it is painful and a bit tricky but I think we should handle this in the following way:
- in the Web app we should have a dropdown selector with both allowed regions (Default, EU) and reuse the region property.
- in the provider check which region is being set and depending on that set the EU URL if that region is selected.

Our design system has already the Dropdown component:
https://github.com/novuhq/novu/blob/8e71c9b09144b5f950c6d8317e2339964051bd93/apps/web/src/design-system/dropdown/Dropdown.tsx#L26

What it might be a bit tricky is in the rendering of the providers is looped so we might need to see how we could identify for the case of Sparkpost that for the Region and type string to render a Dropdown instead an input. We can help you to find the best solution.
Also worth to mention that we are in the process of revamping the integration store so it could happen this change could have conflicts in a near future. But again we can assist with that if needed.

Reply:
Now with the fixes I realised I forgot to explain one important detail.
We should use Region as the field and keep it as string but we also need to accommodate the transition as right now our users have `true` and `false` stored as strings and with the changes their integrations will break.
That means that in the Integrations use cases we would need to transform the boolean as strings values to the proper string to show `EU` if a `true` value is provided. At the very same time that transformation will help that if updating the integration we can slowly replace those `true` strings for the proper `eu` value that should be expected.
In the provider we could keep the boolean check as it would work (for default configuration an empty string or undefined would be stored).
```
      /**
       * Both true and `eu` string would set the first option and false and '' or another non defined value would set the other option
       */
      endpoint: config.eu ? 'https://api.eu.sparkpost.com:443' : undefined, 
``
If this explanation is not very clear I can try to help you to implement that when I have a bit of free time.

Reply:
@p-fernandez I suggest to just write some method like:
```ts
transformRegion(region: string) {
  // ...add some comment to explain why this method exists
  if (region === 'true' || region === true) return 'eu'

  return region;
}

getEndpont(_region: string) {
  const region = this.transformRegion(_region);
  
  switch(region) {
    case 'eu':
      return '...'
    default:
      return;
  }
}
```

I guess some migration script is needed as well.

Reply:
> Now with the fixes I realised I forgot to explain one important detail. We should use Region as the field and keep it as string but we also need to accommodate the transition as right now our users have `true` and `false` stored as strings and with the changes their integrations will break. That means that in the Integrations use cases we would need to transform the boolean as strings values to the proper string to show `EU` if a `true` value is provided. At the very same time that transformation will help that if updating the integration we can slowly replace those `true` strings for the proper `eu` value that should be expected. In the provider we could keep the boolean check as it would work (for default configuration an empty string or undefined would be stored).
> 
> ```
>       /**
>        * Both true and `eu` string would set the first option and false and '' or another non defined value would set the other option
>        */
>       endpoint: config.eu ? 'https://api.eu.sparkpost.com:443' : undefined, 
> ```
> 
> If this explanation is not very clear I can try to help you to implement that when I have a bit of free time.


Actually in `sparkpost.handler.ts` `credentials.region` seems to be string and expected value is `"eu"`, so I decided to leave as is.

https://github.com/novuhq/novu/blob/977b6058d866645faf3c25776f3e1d22a623bdab/packages/application-generic/src/factories/mail/handlers/sparkpost.handler.ts#L18C5-L18C5